provider/src/main/java/com/cloudera/director/google/compute/GoogleComputeProvider.java

```diff
@@ -294,24 +300,39 @@ public void allocate(GoogleComputeInstanceTemplate template,
       // Compose the network url.
       String networkName = template.getConfigurationValue(NETWORK_NAME, templateLocalizationContext);
       String subnetName = template.getConfigurationValue(SUBNETWORK_NAME, templateLocalizationContext);
-      String networkUrl = Urls.buildNetworkUrl(projectId, networkName);
-
-      // Compose the network interface.
-      String accessConfigName = "External NAT";
-      final String accessConfigType = "ONE_TO_ONE_NAT";
-      AccessConfig accessConfig = new AccessConfig();
-      accessConfig.setName(accessConfigName);
-      accessConfig.setType(accessConfigType);
+      String networkProject = template.getConfigurationValue(NETWORK_PROJECT, templateLocalizationContext);
+
+      if (networkProject == null) {
+        networkProject = projectId;
+      }
+
       NetworkInterface networkInterface = new NetworkInterface();
-      networkInterface.setNetwork(networkUrl);
 
       if (subnetName != null) {
         String region = getConfigurationValue(GoogleComputeProviderConfigurationProperty.REGION,
             providerLocalizationContext);
-        networkInterface.setSubnetwork(Urls.buildSubnetUrl(projectId, region, subnetName));
+        String subnetwork = Urls.buildSubnetUrl(networkProject, region, subnetName);
+        networkInterface.setSubnetwork(subnetwork);
       }
+      //  only specify network if the subnetwork isn't set
+      else if (networkName != null && !networkName.equals("default")) {
```

Comment:
Here's another spot where the "default" group is handled specially, and probably shouldn't be.